src/Estimators/include/BipedalLocomotion/Estimators/FloatingBaseEstimator.h

```diff
@@ -0,0 +1,280 @@
+    */
+    virtual const FloatingBaseEstimators::Output& get() const final;
+
+    /**
+    * Determines the validity of the object retrieved with get()
+    * @return True in case of success, false otherwise.
+    */
+    virtual bool isValid() const final { return (m_estimator_state == State::Running); };
+
+    /**
+    * Get ModelComputations object by reference
+    * @return ModelComputations object providing information between considered model related quantities in the estimator
+    * like the base link, IMU, feet contact frames.
+    */
+    ModelComputations& modelComputations();
+
+protected:
+    /**
+    * Configure generic parameters
+    * @param[in] handler configure the generic parameters for the estimator
+    * @return True in case of success, false otherwise.
+    */
+    bool initialize(std::weak_ptr<BipedalLocomotion::ParametersHandler::IParametersHandler> handler);
+
+    /**
+    * These custom parameter specifications should be specified by the derived class.
+    * @param[in] handler configure the custom parameters for the estimator
+    * @return bool
+    */
+    virtual bool customInitialization(std::weak_ptr<BipedalLocomotion::ParametersHandler::IParametersHandler> handler) { return true; };
+
+    /**
+    * Propagate the states through the prediction model, if there exists any (eg. a strap-down IMU model)
+    * @param[in] meas measurements passed as exogeneous inputs to the prediction model
+    * @param[in] dt sampling period in seconds
+    * @param[out] state previous state estimate
+    * @param[out] P previous state covariance matrix
+    * @return True in case of success, false otherwise.
+    */
+    virtual bool predictState(const FloatingBaseEstimators::Measurements& meas,
+                              const double& dt) { return true; };
+
+    /**
+    * Update the predicted state estimates using kinematics and contact measurements
+    * @param[in] meas measurements to update the predicted states
+    * @param[in] dt sampling period in seconds
+    * @return True in case of success, false otherwise.
+    */
+    virtual bool updateKinematics(const FloatingBaseEstimators::Measurements& meas,
+                                  const double& dt) { return true; };
+
+   /**
+   * Update the states and the associated covariance using the Kalman gain and the innovations (in an EKF based implementation)
+   * @note this function should be called from within updateKinematics in an EKF base implementation, otherwise left unused
+   * @param[in] innovation innovation vector
+   * @param[in] measurement_model_jacobian discrete-time measurement model Jacobian
+   * @param[in] measurement_noise_var discrete-time measurement noise covariance matrix
+   * @return bool
+   */
+   virtual bool updateStates(const Eigen::VectorXd& innovation,
+                             const Eigen::MatrixXd& measurement_model_jacobian,
+                             const Eigen::MatrixXd& measurement_noise_var) { return true; };
```

Comment:
I will remove this method from the base class. It does not extend very well across different derived estimators.

Reply:
This will allow us to maintain only high-level measurement update commands from the base class. eg. updateKinematics().

Reply:
Done in https://github.com/dic-iit/bipedal-locomotion-framework/pull/79/commits/c401bc47c72623de80ab17f291bfed9461bbb76e